op-service/eth/types.go

```diff
@@ -293,3 +300,11 @@ type SystemConfig struct {
 	GasLimit uint64 `json:"gasLimit"`
 	// More fields can be added for future SystemConfig versions.
 }
+
+// Withdrawal represents a validator withdrawal from the consensus layer.
+type Withdrawal struct {
```

Comment:
why do you need to paste this struct here rather than just using types.Withdrawal to use the one already defined by op-geth?

Reply:
We did have a bug because of this previously when the withdrawals were added we didn't. know how to deserialize

Reply:
My thinking was mostly to stick with the current convention of duplicating. e.g. [ExecutionPayload - optimism](https://github.com/ethereum-optimism/optimism/blob/develop/op-service/eth/types.go#L128) and [ExecutableData - op-geth](https://github.com/ethereum-optimism/op-geth/blob/optimism/beacon/engine/types.go#L60).

I think decoupling the client/server types keeps it a bit safer, i.e. if we have some reflection somewhere and an additional field gets added and breaks some encoding.